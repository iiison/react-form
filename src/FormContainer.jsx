import React, { Component } from 'react'
import PropTypes            from 'prop-types' // eslint-disable-line import/no-extraneous-dependencies

import validations          from './utils/validationRules'

/**
 * Select rule from 3 diffrent places(predefined rules, form level rules or input level rules)
 */
export function selectRule({ ruleValue, customFormRules, customRules }) {
  return validations[ruleValue] || customRules[ruleValue] || customFormRules[ruleValue]
}

/**
 * Validates if the the selected rule(returned from `selectRule`) is valid or not.
 * if rule is valid then tests the input field value agains the selected rule.
 * returns empty string if rule is valid and saticefies the condition.
 */
export function validateRule({ id, value, displayName, ruleValue, ruleArgs, validation }) {
  if (validation) {
    const stringifiedValue = value.toString()
    const isRuleSatisfied = ( ruleValue !== 'required' && !stringifiedValue )
      ? true
      : validation.rule.apply(null, ruleArgs).test(value.toString())
    let error = ''

    if (!isRuleSatisfied) {
      error = validation.formatter.apply(null, [displayName || id, ...ruleArgs])

      return error
    }
  } else {
    throw `invalid validation rule: ${ruleValue}, please use an existing validation rule name or pass a custom function with same name through 'customRules' prop in Input: ${id}. Rule value should be an object with keys: 'rule' as an Regex and 'formatter' as a function, that formats the value.` // eslint-disable-line
  }

  return ''
}


/**
 * Main Component, a HOC, will store all fields' state.
 */
export default class FormContainer extends Component {
  constructor(props) {
    super(props)

    const {
      isDisabled = false,
      shouldValidateForm,
      defaultClasses
    } = this.props

    this.state = {
      shouldValidateForm,
      defaultClasses,
      isDisabled,
      errors     : {},
      isFetching : false
    }
  }

  getChildContext() {
    return {
      setFieldValue : ({ event, field, value, isMultipleValues, id = 'value' }, callBack) => {
        const fieldName = field.id
        const setState = (name, fieldValue) => {
          this.setState((prevState) => ({
            ...prevState,
            fields : {
              ...prevState.fields,
              [fieldName] : {
                ...prevState.fields[name],
                shouldValidateField : true,
                [id]                : fieldValue
              }
            }
          }), callBack)
        }

        if (!value && value !== '') {
          setState(fieldName, event.currentTarget.value)
        } else if (isMultipleValues) {
          const stateCopy = { ...this.state }

          for (const item in value) {
            stateCopy.fields[item].shouldValidateField = true
            stateCopy.fields[item][id] = value[item]
          }

          this.setState((prevState) => ({
            ...prevState,
            ...stateCopy
          }))
        } else {
          setState(fieldName, value)
        }

        return this.state
      },

      addField : (data) => {
        this.setState((prevState) => ({
          ...prevState,
          errors : {
            ...prevState.errors,
            [data.id] : ''
          },
          fields : {
            ...prevState.fields,
            [data.id] : {
              ...data,
              // shouldValidateField : false // Not sure why this was added
            }
          }
        }))
      },

      setFormData : (data) => {
        this.setState((prevState) => ({
          ...prevState,
          ...data
        }))
      },

      validateForm : (fieldName) => {
        const { shouldValidateForm, fields } = this.state
        const field = fields[fieldName]

        if (shouldValidateForm) {
          if (field) {
            if ( field.shouldValidateField ) {
              this.validateField(field)
            }

            return
          }

          for (const unit in fields) {
            const fieldData = fields[unit]

            if (fieldData) {
              this.validateField(fieldData)
            }
          }
        }

        return
      },
      formData : this.state
    }
  }

  /**
   * React Lifecycle Method: Renders the data
   *
   * @return {DOM} Main container DOM.
   */
  render() {
    const { children } = this.props

    return (
      <form action=''>
        {children}
      </form>
    )
  }

  validateField = ({
    id,
    value,
    validate,
    displayName,
    customRules = {}
  }) => {
    let error = ''
    const rules = validate ? validate.split('|') : ''
    const { customValidators : customFormRules } = this.props

    if (rules.length) {
      for (const rule in rules) {
        const ruleName = rules[rule]
        const ruleDetails = ruleName.split('-')
        const [ruleValue, ...ruleArgs] = ruleDetails
        const validation = selectRule({ ruleValue, customFormRules, customRules })

        try {
          error = validateRule({ id, value, displayName, ruleValue, ruleArgs, validation })

          if (error !== '') {
            break
          }
        } catch (invalidRuleError) {
          throw invalidRuleError
        } 
      }

      this.setState((prevState) => ({
        ...prevState,
        errors : {
          ...prevState.errors,
          [id] : error
        }
      }))
    }
  }

  static defaultProps = {
    errors             : {}, // eslint-disable-line
    children           : <div />,
    isDisabled         : false,
    shouldValidateForm : true,
    customValidators   : {},
    defaultClasses     : {
      contClass  : '',
      fieldClass : '',
      errorClass : '',
      labelClass : ''
    }
  }

  static propTypes = {
    children           : PropTypes.node,
    isDisabled         : PropTypes.bool,
    shouldValidateForm : PropTypes.bool,
    customValidators   : PropTypes.object, // eslint-disable-line
    defaultClasses     : PropTypes.shape({
      labelClass : PropTypes.string,
      contClass  : PropTypes.string,
      errorClass : PropTypes.string,
      fieldClass : PropTypes.string,
    })
  }

  static childContextTypes = {
    addField      : PropTypes.func,
    setFieldValue : PropTypes.func,
    validateForm  : PropTypes.func,
    formData      : PropTypes.object, // eslint-disable-line
    setFormData   : PropTypes.func
  }
}

