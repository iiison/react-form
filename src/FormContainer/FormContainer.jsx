import React, { Component } from 'react'

import { Form } from '$COMPONENTS'
import validations from '$UTILS/validationRules'

/**
 * Test all validation rules ofa field
 *
 * @param {Object} fieldData all field realated data
 *
 * @return {String}         validation error ina field
 */
export function testFieldRules(fieldData) {
  const { validate, value, displayName } = fieldData

  const validationRules = validate.split('|')

  for (const validation in validationRules) {
    const { rule, formatter } = validations[validationRules[validation]]()

    const result = rule.test(value)

    if (result === false) {
      return formatter(displayName)
    }
  }
}

/**
 * Generic Form Handler, Handles form state,
 * Pass Form elemnt's JSON throught the FormContainer and
 * it will redner the form.
 */
export default class FormContainer extends Component {
  /**
   * Constructor for the container
   *
   * @param {Object} props form data
   */
  constructor(props) {
    super(props)
    const { formData : { fields } } = props

    this.state = {
      isTested : false,
    }

    this.state.fields = fields.reduce((prev, curr) => {
      const temp = {
        value          : '',
        error          : '',
        shouldBeTested : false,
        ...curr
      }

      return {
        ...prev,
        [temp.id] : temp
      }
    }, {})
  }

  /**
   * DOM Event : Chaanges `shouldBeTested`, will be executed on
   * keyUp in an input field
   *
   * @param {String} field field name
   */
  makeFieldTestable = (field) => {
    const fieldRef = this.state.fields[field]

    fieldRef.shouldBeTested = true

    this.setState((previousState) => {
      return {
        fields : {
          ...previousState.fields,
          [field] : fieldRef
        }
      }
    })
  }

  /**
   * Handle Key press event on the input,
   * tests field value if field is already has error.
   * @param  {String} field Field name in which event was started
   */
  handleKeyPress = (field) => {
    const fieldData = this.state.fields[field]
    const { error } = fieldData

    this.makeFieldTestable(field)

    if (error) {
      this.validateInput(field)
    }
  }

  /**
   * DOM Event: Handles input value change
   *
   * @param {Object} event DOM event object
   */
  handleChange = (event) => {
    event.preventDefault()

    const elemRef = event.target
    const idName = elemRef.id

    this.setState((previousState) => {
      const { fields } = previousState

      return {
        fields : {
          ...fields,
          [idName] : {
            ...fields[idName],
            value : elemRef.value
          }
        }
      }
    })
  }

  /**
   * Validate single input
   *
   * @param  {String}  field       field name to be tested
   * @param  {Boolean} isFinalTest will be true if whole form is being tested
   * @return {String}              validation error of the field
   */
  validateInput = (field, isFinalTest) => {
    const fieldData = this.state.fields[field]
    const { validate } = fieldData

    if ((fieldData.shouldBeTested === true || isFinalTest) && validate) {
      const error = testFieldRules(fieldData) || ''

      this.setState((previousState) => {
        const { fields } = previousState
        const updatedField = {
          ...fields[field],
          error
        }

        return {
          fields : {
            ...fields,
            [field] : updatedField
          }
        }
      })

      return error
    }
  }

  /**
   * Handles Form Submission, callback actual form submit fxn
   * after the form validation. validation is the part of plugin.
   *
   * @param {Object} event DOM Event object
   */
  handleFormSubmit = (event) => {
    event.preventDefault()

    const { handleSubmit } = this.props
    const { fields } = this.state
    const errors = []

    for ( const field in fields ) {
      const error = this.validateInput(field, true)

      error && errors.push(error)
    }

    if (!errors.length) {
      handleSubmit(fields)
    }
  }

  /**
   * React Lifecycle Method: Renders the data
   *
   * @return {DOM} Main container DOM.
   */
  render() {
    const { fields } = this.state
    const {
      error,
      messages,
      isFetching,
      submitButtonText,
      submitButtonClassName } = this.props

    const props = {
      fields,
      messages,
      events : {
        onSubmit          : this.handleFormSubmit,
        onValueChange     : this.handleChange,
        validateInput     : this.validateInput,
        handleKeyPress    : this.handleKeyPress,
        makeFieldTestable : this.makeFieldTestable
      },
      submitButtonData : {
        error,
        isFetching,
        submitButtonText,
        submitButtonClassName,
        onSubmit : this.handleFormSubmit
      }
    }

    return (
      <Form
        {...props}
      />
    )
  }

  /**
  * React Lifecycle Method: Executes right before the
  * component mounts
  */
  componentWillMount() {
  }
}
