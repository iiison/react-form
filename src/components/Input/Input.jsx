import React, { Component } from 'react'
import PropTypes from 'prop-types' // eslint-disable-line import/no-extraneous-dependencies

export default class Input extends Component {
  render() {
    const {
      id,
      type,
      error,
      value,
      label,
      placeholder,
      onFieldChange,
      classes :  {
        contClass = '',
        inputClass = '',
        errorClass = '',
        labelClass = ''
      } = {}
    } = this.props
    const {
      setFieldValue,
      formData = {}
    } = this.context
    const field = formData.fields ? formData.fields[id] : {}

    return (
      <div className={`${contClass} input-cont col-12 grid`}>
        {
          label
            ? <div className={`col-12 ${labelClass} label`}>{label}</div>
            : ''
        }
        <input
          type={type}
          placeholder={placeholder}
          value={field.value || value}
          className={`${inputClass} col-12`}
          onChange={(evt) => {
            const event = { ...evt }

            evt.persist()
            setFieldValue({
              event,
              field
            })

            if (onFieldChange && typeof onFieldChange === 'function') {
              onFieldChange(event, field, setFieldValue)
            }
          }}
        />
        {
          error
            ? <div className={`col-12 error ${errorClass}`}>{error}</div>
            : ''
        }
      </div>
    )
  }

  componentDidMount() {
    this.context.addField(this.props)
  }

  static contextTypes = {
    addField      : PropTypes.func.isRequired,
    setFieldValue : PropTypes.func.isRequired,
    formData      : PropTypes.object.isRequired
  }

  static propTypes = {
    id            : PropTypes.string.isRequired,
    type          : PropTypes.string,
    error         : PropTypes.string,
    value         : PropTypes.string,
    label         : PropTypes.string,
    classes       : PropTypes.object,
    validate      : PropTypes.string,
    placeholder   : PropTypes.string,
    onFieldChange : PropTypes.func
  }

  static defaultProps = {
    type          : 'text',
    validate      : null,
    value         : '',
    onFieldChange : null
  }
}

