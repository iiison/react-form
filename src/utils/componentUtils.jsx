import React from 'react'

export function drawElements ({ shouldUseDefaultClasses, classes, content, defaultClasses }) {
  return content
    ? <div className={`col-12 ${classes} ${shouldUseDefaultClasses && defaultClasses}`}>{content}</div>
    : ''
}

