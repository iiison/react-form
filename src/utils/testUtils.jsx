import React, { Component } from 'react'

export function wrapWithContext(context, contextTypes, Elements) {
  class WrapWithContext extends Component {
    getChildContext = () => context

    render() {
      return (
        <div>
          {this.props.children}
        </div>
      ) 
    }

    getChildContext() {
      return context
    }

    static childContextTypes = contextTypes
  }

  return WrapWithContext
}

