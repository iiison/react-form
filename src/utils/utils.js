/**
 * Debounce input function
 *
 * @param   {Function} func function to be debounced
 * @param   {Number}   time time to wait
 *
 * @returns {Function}      debounced function
 */
export function debounce(func, time) {
  let timeout

  return function(...args) {
    const ref = this

    clearTimeout(timeout)
    timeout = setTimeout(() => func.apply(ref, args), time)
  }
}

