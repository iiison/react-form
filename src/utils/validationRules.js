const validations = {
  email() {
    return {
      rule    : /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i,
      formatter (fieldName) {
        const message = '%c is not valid'

        return message.replace('%c', fieldName)
      }
    }
  },
  required() {
    return {
      rule    : /\S/,
      formatter (fieldName) {
        const message = '%c is required'

        return message.replace('%c', fieldName)
      }
    }
  }
}

export default validations
