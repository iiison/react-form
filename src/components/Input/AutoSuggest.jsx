import React, { Component } from 'react'
import PropTypes from 'prop-types' // eslint-disable-line import/no-extraneous-dependencies

import { debounce } from '../../utils/utils'
import Input from './Input'

function renderSuggestions ({ Template, state, suggestionsClasses, clickEvent }) {
  const { currentIndex, suggestions, error, showSuggestions } = state
  const notFoundMessage = 'No matching suggestion.'
  let items

  if (!showSuggestions) {
    return ''
  }

  if (error || !suggestions.length) {
    items = <li className={'suggestions col-12 lighter-color'}>{'No matching suggestion.'}</li>
  }

  items = suggestions.map((suggestion, index) => {
    const isSelected = currentIndex === index
    const props = {
      ...suggestion,
      isSelected,
    }

    return (
      <Template
        {...props}
        key={index} // eslint-disable-line react/no-array-index-key
        clickEvent={() => clickEvent(suggestion)}
      />
    ) 
  })

  return (<ul className={`col-12 grid suggestions ${suggestionsClasses.contClass}`}>{items}</ul>)
}

function getIndexOnArrowPress(currentIndex, key, suggestions) {
  const totalSuggestions = suggestions.length - 1
  const arrowKeyIndexMap = {
    ArrowDown : 1,
    ArrowUp   : -1
  }
  const index = currentIndex + arrowKeyIndexMap[key]
  let newIndex = index

  if (newIndex < 0) {
    newIndex = totalSuggestions
  }

  if (newIndex > totalSuggestions) {
    newIndex = 0
  }
  
  return newIndex
}

export default class AutoSuggest extends Component {
  state = {
    showSuggestions : false,
    currentIndex    : 0,
    suggestions     : [],
    error           : ''
  }

  setCurrentIndex = (index) => {
    this.setState({
      currentIndex : index
    })
  }

  handleInputBlur = (event) => {
    const { onBlur } = this.props.events

    this.setState({
      showSuggestions : false
    })
    
    if (onBlur && typeof onBlur === 'function') {
      onBlur(event)
    }
  }

  handleKeyDown = ({ key }) => {
    const { onValueSelection } = this.props
    const { currentIndex, suggestions, showSuggestions } = this.state

    switch (key) {
    case 'Escape' : 
      this.setState({ showSuggestions : false })
      break

    case 'ArrowDown' :
    case 'ArrowUp' :
      const newIndex = getIndexOnArrowPress(currentIndex, key, suggestions)

      this.setState({
        currentIndex    : newIndex,
        showSuggestions : true
      })
      break

    case 'Enter' :
      if (showSuggestions && suggestions.length) {
        onValueSelection(suggestions[currentIndex])
        this.setState({ showSuggestions : false })
      }
      break
    default : 
      return
    }
  }

  handleSuggestionClick = (suggestion) => {
    this.props.onValueSelection(suggestion)
    this.setState({
      showSuggestions : false
    })
  }

  handleFieldChange = async (event, field, setFieldValue) => {
    this.setState({ showSuggestions : true })

    const { onFieldChange, getSuggestions, minLength } = this.props
    const value = event.target.value

    if (!value) {
      this.setState({
        suggestions     : [],
        showSuggestions : false
      })

      return
    }

    if (value && value.length > minLength) {
      const { suggestions = [], error } = await getSuggestions(value)

      this.setState({
        error,
        suggestions
      })

      if (onFieldChange && typeof onFieldChange === 'function') {
        onFieldChange(event, field, setFieldValue)
      }

      return
    }
  }

  render() {
    const state = this.state
    const clickEvent = this.handleSuggestionClick
    const { Template, suggestionsClasses, ...rest } = this.props

    rest.onFieldChange = this.debouncedHandleChange
    rest.events = {
      onKeyDown : this.handleKeyDown,
      // onBlur    : this.handleInputBlur
    }

    return (
      <div className={`col-12 grid relative`}>
        <Input {...rest} />
        {renderSuggestions({ Template, state, suggestionsClasses, clickEvent })}
      </div>
    )
  }

  componentWillMount() {
    this.debouncedHandleChange = debounce(this.handleFieldChange, this.props.delay)
  }

  static defaultProps = {
    delay              : 500,
    minLength          : 3,
    events             : {},
    suggestionsClasses : {
      contClass : ''
    }
  }

  static contextTypes = {
    setFieldValue : PropTypes.func.isRequired,
    formData      : PropTypes.object.isRequired
  }

  static propTypes = {
    onFieldChange      : PropTypes.func, // eslint-disable-line react/require-default-props
    getSuggestions     : PropTypes.func.isRequired,
    onValueSelection   : PropTypes.func.isRequired,
    Template           : PropTypes.func.isRequired,
    minLength          : PropTypes.number,
    delay              : PropTypes.number,
    suggestionsClasses : PropTypes.shape({
      contClass : PropTypes.string
    })
  }
}

