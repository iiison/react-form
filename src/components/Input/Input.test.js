import React               from 'react'
import renderer            from 'react-test-renderer'
import PropTypes           from 'prop-types' // eslint-disable-line import/no-extraneous-dependencies
import { mount }           from 'enzyme'

import Input               from './Input'
import FormContainer       from '../../FormContainer'
import { wrapWithContext } from '../../utils/utils'

/* eslint-disable react/jsx-filename-extension */
const defaultContext = {
  addField      : jest.fn(),
  setFieldValue : jest.fn(),
  validateForm  : jest.fn(),
  formData      : {
    fields : {
      email : {
        value : ''
      }
    }
  }
}
const contextTypes = {
  addField      : PropTypes.func,
  setFieldValue : PropTypes.func,
  formData      : PropTypes.object,
  validateForm  : PropTypes.func,
}
const defaultInputArgs = {
  id            : 'email',
  type          : 'text',
  value         : '',
  onFieldChange : null,
  validate      : 'required',
  placeholder   : 'enter value',
  displayName   : 'Email',
  classes       : {
    contClass  : 'cont-class',
    fieldClass : 'field-class',
    errorClass : 'error-class',
    labelClass : 'label-class'
  },
}

// Snapshot matching for Login Container
describe('>>> Input Container -- Snapshot Test', () => {
  it('Matches the snapshot with default props', () => {
    debugger
    const context = {
      ...defaultContext,
      fields : {
        ...defaultContext.fields,
        email : { ...defaultInputArgs }
      }
    }
    const ComponentWithContext = wrapWithContext(defaultContext, contextTypes, Element)
    const tree = renderer.create(<ComponentWithContext><Input value='' id='email' /></ComponentWithContext>).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('Matches the snapshot with predefined values', () => {
    // const ComponentWithContext = wrapWithContext(defaultContext, contextTypes, Element)
    // const tree = renderer.create(
    //   <ComponentWithContext>
    //     <Input id='email' value='test' />
    //   </ComponentWithContext>
    // ).toJSON()
    //
    // expect(tree).toMatchSnapshot()
  })
})
// *************************************************************

/* eslint-enable */


// Check DOM Behaviour
{/* describe('>>> Input Container -- Shallow Rendering', () => { */}
{/*   let mountedScreen */}
{/*   const inputScreen = (customProps = {}, customContext = {}) => { */}
{/*     const context = { */}
{/*       ...defaultContext, */}
{/*       ...customContext */}
{/*     } */}
{/*     const ComponentWithContext = wrapWithContext(context, contextTypes, Element) */}
{/*     const initialState = { */}
{/*       id    : 'email', */}
{/*       value : '' */}
{/*     } */}
{/*  */}
{/*     if (!mountedScreen) { */}
{/*       const updatedState = { */}
{/*         ...initialState, */}
{/*         ...customProps */}
{/*       } */}
{/*  */}
{/*       mountedScreen = mount(<ComponentWithContext><Input {...updatedState} /></ComponentWithContext>) */}
{/*     } */}
{/*  */}
{/*     return mountedScreen */}
{/*   } */}
{/*  */}
{/*   beforeEach(() => { */}
{/*     mountedScreen = undefined */}
{/*   }) */}
{/*  */}
{/*   it('Renders dumb component with default props', () => { */}
{/*     const tree = inputScreen() */}
{/*     const expectedValue = '' */}
{/*     const resultValue = tree */}
{/*       .find('input') */}
{/*       .at(0) */}
{/*       .props() */}
{/*       .value */}
{/*  */}
{/*     expect(resultValue).toEqual(expectedValue) */}
{/*     expect(defaultContext.addField).toHaveBeenCalledWith(defaultInputArgs) */}
{/*   }) */}
{/*  */}
{/*   it('Renders input with preset value', () => { */}
{/*     const props = { */}
{/*       value : 'test', */}
{/*     } */}
{/*     const tree = inputScreen(props) */}
{/*  */}
{/*     const expectedValue = 'test' */}
{/*     const resultValue = tree */}
{/*       .find('input') */}
{/*       .at(0) */}
{/*       .props() */}
{/*       .value */}
{/*  */}
{/*     expect(resultValue).toEqual(expectedValue)  */}
{/*   }) */}
{/*  */}
{/*   it('Triggers input change function when input changed', () => { */}
{/*     const props = { */}
{/*       onFieldChange : jest.fn() */}
{/*     } */}
{/*     const tree = inputScreen(props) */}
{/*  */}
{/*     const input = tree */}
{/*       .find('input') */}
{/*       .at(0) */}
{/*  */}
{/*     input.simulate('change', { */}
{/*       target : { */}
{/*         value : 'test' */}
{/*       } */}
{/*     }) */}
{/*  */}
{/*     expect(props.onFieldChange).toHaveBeenCalled() */}
{/*     expect(defaultContext.setFieldValue).toHaveBeenCalled() */}
{/*   }) */}
{/* }) */}
// *************************************************************

