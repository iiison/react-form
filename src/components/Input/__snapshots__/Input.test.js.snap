// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> Input Container -- Snapshot Test Matches the snapshot when Textarea is rendered 1`] = `
<form
  action=""
>
  <div
    className="cont-class   input-cont col-12 grid"
  >
    
    <textarea
      className="field-class  col-12"
      defaultValue="big textarea value"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="custom textarea placeholder"
      rows={5}
    />
    
  </div>
</form>
`;

exports[`>>> Input Container -- Snapshot Test Matches the snapshot with custom props 1`] = `
<form
  action=""
>
  <div
    className="cont-class   input-cont col-12 grid"
  >
    
    <input
      className="field-class  col-12"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="custom placeholder"
      type="text"
      value=""
    />
    
  </div>
</form>
`;

exports[`>>> Input Container -- Snapshot Test Matches the snapshot with default props 1`] = `
<form
  action=""
>
  <div
    className="   input-cont col-12 grid"
  >
    
    <input
      className="  col-12"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value=""
    />
    
  </div>
</form>
`;

exports[`>>> Input Container -- Snapshot Test Matches the snapshot with predefined value 1`] = `
<form
  action=""
>
  <div
    className="   input-cont col-12 grid"
  >
    
    <input
      className="  col-12"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
      value="test"
    />
    
  </div>
</form>
`;
